examples/web/nextjs/pages/_app.js

```diff
@@ -0,0 +1,48 @@
+// :snippet-start: custom-app-wrapper
+import { useMemo } from "react";
+import * as Realm from "realm-web";
+import Layout from "../components/layout"; // :remove:
+// :state-start: ssr
+import { setCookie } from "nookies";
+// Import the useEffect hook
+import { useEffect } from "react";
+// :state-end:
+
+function MyApp({ Component, pageProps }) {
+  // useMemo with empty dependency array only computes value for `app` on mount
+  const app = useMemo(() => new Realm.App(process.env.NEXT_PUBLIC_APP_ID), []);
```

Comment:
Still not clear why we need this useMemo, especially in the non-ssr state where `app` is an unused variable. Could we:

- Remove the useMemo and just get the app directly
- move the call into the `ssr` state OR move it out of the component entirely

Reply:
this just guarantees that there's always an app client instantiated in the app. 

i guess you're right that it's not strictly nec here. there was an earlier version of this guide where it was more important. lemme spend a lil more time looking at.

Reply:
I think either of my suggestions in the second bullet would also guarantee that it exists

Reply:
the first would only guarantee it exists given that the SSR state exists. 

for the second, not sure what moving it out of the component would add. 

after thinking about a bit more, what i like about this implementation is that if there is already a logged in user, the user reauthenticates automatically when this component is rendered.

Reply:
This should be equivalent (and you could even move it out of your component's render and into its own file):

```suggestion
  const app = Realm.getApp(process.env.NEXT_PUBLIC_APP_ID);
```

There should be little reason to confuse users by mixing between `Realm.getApp` (or the longer `Realm.App.getApp`) and the `new Realm.App(...)` syntax. I'd suggest picking one and sticking to it.